graphql_compiler/schema.py

```diff
@@ -349,3 +349,17 @@ def insert_meta_fields_into_existing_schema(graphql_schema):
                                      .format(meta_field_name))
 
             type_obj.fields[meta_field_name] = meta_field
+
+
+def _check_for_nondefault_directives(directives):
+    """Check if any user-created directives are present"""
+    deprecated_directive_names = {'deprecated', 'skip', 'include'}
+    default_directive_names = {directive.name for directive in DIRECTIVES}
```

Comment:
Might be a good idea to check these for equality, instead of just comparing names. It's possible that someone defines a different `@optional` directive for example. Not likely to happen, but if it does it will be difficult to find because we think we are addressing the problem.

Reply:
Directive definitions include a description, but the argument directives do not. If we check for equality, the descriptions get compared as well. 
What if we just create a copy of DIRECTIVES, with descriptions nulled out? What about deprecated, skip, and include then?
It's way too adversarial to assume users will create their own directives, with the same name as the ones we've reserved.

Reply:
Changed the function name to reflect it only compares names.
Get_schema_for_macro_definitons now has a precondition assuming all directives we support like @fold, @output, etc. have their default behavior.